react/BottomSheet/README.md

```diff
@@ -0,0 +1,42 @@
+Displays content coming up from the bottom of the screen. The pane can be swiped to the top of the screen.
+
+```jsx
+import BottomSheet, { BottomSheetContentBlock } from 'cozy-ui/transpiled/react/BottomSheet'
+import Button from 'cozy-ui/transpiled/react/Buttons'
+
+// <-- only usefull for the documentation
+initialState = { isBottomSheetDisplayed: isTesting() }
+const showBottomSheet = () => setState({ isBottomSheetDisplayed: true })
+
+const settings = { mediumHeightRatio: isTesting() ? 0.90 : 0.33 }
+const classes = { header: 'u-ph-1 u-pb-1' }
+// -->
+
+;
+
+<>
+  <Button label="Open BottomSheet" onClick={showBottomSheet} />
+  {state.isBottomSheetDisplayed && (
+    <BottomSheet
+      classes={classes}
+      header={
+        <>
+          <Button className="u-mr-half" variant="secondary" label="Button 1" fullWidth />
+          <Button variant="secondary" label="Button 2" fullWidth />
+        </>
+      }
+      content={
+        <>
+          <BottomSheetContentBlock>
+            {content.ada.short}
+          </BottomSheetContentBlock>
+          <BottomSheetContentBlock disableElevation>
+            {content.ada.long}
+          </BottomSheetContentBlock>
+        </>
+      }
+      settings={settings}
+    />
```

Comment:
I don't like the API, it doesn't follow the standard we try to implement, no? It seems we were leaving the "props component" in UI. MUI do not provide something like: 
```
<List
items={...}
/>
```

but instead:
```
<List>
<ListItem>
...
```

Don't we want something composable, like our dialogs for instance? 

```
<BottomSheet>
   <BottomSheetHeader />
   <BottomSheetContent /> 
   <BottomSheetFooter />
</BottomSheet>
```

Reply:
Also, the plan is to use this BottomSheet everywhere where we use the actual ActionMenu / Menu https://github.com/cozy/cozy-ui/issues/1848

And the closet the APIs are, easiest will be the migration. 

(If you can keep the same API and the same component in our app but below cozy-ui renders a BottomSheet for an ActionMenu it'll be perfect)

Reply:
> it doesn't follow the standard we try to implement, no?

> we want something composable, like our dialogs

Nos `CozyDialogs` (composant haut niveau intelligent) ne sont pas composables comme tu l'indiques, il y a des props `title`, `content` et `actions` qui permettent d'injecter le contenu, mais on ne peut pas en modifier la structure. C'est sur ce modèle que je me suis calqué pour faire le BottomSheet, modèle qui me semblait être notre standard justement pour les composants haut niveau.

On a aussi les `Dialogs`, composables en effet, mais ce n'est que du Mui skinné. Et on est censé les utiliser uniquement pour un cas particulier non gérable avec les `CozyDialogs`, ce qui a priori n'est pas censé arriver.

Le `Viewer` dans son genre (plus vieux donc pas tout à fait le même type d'API) n'est pas composable non plus, c'est aussi un haut niveau qui attend son contenu via des props.

> the plan is to use this BottomSheet everywhere where we use the actual ActionMenu / Menu

Effectivement le `ActionMenu` est composable.
Concernant ce composant, d'un part je pense qu'on va mettre longtemps à valider la PR si on tente de résoudre 2 problèmes d'un seul coup (le bottomSheet compatible Viewer + App ainsi que remplacer le comportement du ActionMenu - est-ce qu'on veut faire ça en 2 temps ou non, si non est-ce qu'on veut prendre le temps maintenant ?), d'autre part je pensais rester sur le même modèle, à savoir un composant haut niveau avec une API simple (des props `header`, `content` etc.) ainsi faire un BC simple puisque sur ce modèle, c'est toujours l'app qui injecte le contenu, c'est juste la coquille, qui est pré-mâchée en somme, qui change.

Il y a quand même une question importante à résoudre ici : **quel est le standard qu'on veut suivre ?**

Si on fait le parallèle avec l'atomic design, pour moi cozy-ui offre des atomes (bas niveau) et des organismes (haut niveau). Je n'avais pas en tête que ces derniers soient modulables. Sinon qu'en est-il des `CozyDialogs`, du `Viewer`, de `SquareAppIcon` et probablement d'autres ?

Reply:
Petit commit en plus pour la version composable :) (plus proche de l'adn Mui)